File: easy/black_card.rb
File.open(ARGV[0]).each_line do |line|
  s = line.chomp.split(' | ')
  t = s[0].split
  m = s[1].to_i
  while t.length > 1
    n = m % t.length - 1
    t.delete_at((n >= 0) ? n : t.length - 1)
  end
  puts t[0]
end

File: easy/black_card.pl
use strict;
open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $line = <INFILE> ) {
    chomp($line);
    my ( $s, $m ) = split( / \| /, $line );
    my @t = split( / /, $s );
    while ( scalar @t > 1 ) {
        my $n = $m % ( scalar @t ) - 1;
        $n >= 0 ? splice( @t, $n, 1 ) : pop @t;
    }
    print "@t[0]\n";
}
close(INFILE);
